indigo_api/models/works.py

```diff
@@ -455,9 +456,13 @@ def versions(self):
         """
         return Version.objects.get_for_object(self).select_related('revision', 'revision__user')
 
-    def as_at_date(self):
+    def computed_as_at_date(self):
```

Comment:
I would do this the other way -- callers shouldn't have to care that this is computed, they call `as_at_date` as before (ie. don't change the object's API). Just use a field name that doesn't clash, eg. `as_at_date_override`? That means you need to change less of the code that makes use of this existing method.